neurow/lib/neurow/receiver_shard_manager.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Neurow.ReceiverShardManager do
```

Comment:
Maybe it should be a [DynamicSupervisor](https://hexdocs.pm/elixir/DynamicSupervisor.html) ? So ReceiverShard will be children of the ReceiverShardManager in the supervision tree?